src/boot/efi/boot.c

```diff
@@ -137,13 +136,19 @@ static BOOLEAN line_edit(
                 }
                 print[j] = '\0';
 
+                /* See comment at edit_line() call site for why we start at 1. */
                 print_at(1, y_pos, config->edit_color, print);
-                /* See comment at edit_line() call site. */
-                uefi_call_wrapper(ST->ConOut->SetCursorPosition, 3, ST->ConOut, cursor + 1, y_pos);
 
-                err = console_key_read(&key, 0);
-                if (EFI_ERROR(err))
-                        continue;
+                if (!print[cursor])
+                        print[cursor] = ' ';
+                print[cursor+1] = '\0';
+                do {
+                        print_at(cursor + 1, y_pos, cursor_color, print+cursor);
+                        cursor_color = TEXT_ATTR_SWAP(cursor_color);
+
+                        err = console_key_read(&key, 750);
+                        print_at(cursor + 1, y_pos, config->edit_color, print+cursor);
+                } while (EFI_ERROR(err));
```

Comment:
i'd be super curius how this actually looks like, any chance you can attach a screenshot of the relevant UI changes?